tests: msgq_usage: add semaphores to ensure all services started before client query

IRISZZW opened the pull request:
add semaphores to ensure all services started before client query.
otherwise client query services may fail.

Fix: #35215

abrodkin commented:
Even if the problem is in a service starting a bit late, I don't like this solution with a random delay insertion as it might work for HSDK but be not long enough for some other platforms. That said IMHO it worth having some real synchronization primitive here.

IRISZZW committed:
tests: msgq_usage: ensure all services started before client query

add semaphores to ensure all services started before client query.
otherwise client query services may fail.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

IRISZZW commented:
> Even if the problem is in a service starting a bit late, I don't like this solution with a random delay insertion as it might work for HSDK but be not long enough for some other platforms. That said IMHO it worth having some real synchronization primitive here.
Thanks for your review, I add semaphores to ensure all services started before client query
